src/main/java/me/qoomon/gitversioning/commons/GitUtil.java

```diff
@@ -78,7 +78,9 @@ public static GitDescription describe(ObjectId revObjectId, Pattern tagPattern,
                 throw new IllegalStateException("couldn't find matching tag in shallow git repository");
             }
 
-            return new GitDescription(revObjectId.getName(), "root", depth);
+            // If no matching tag found, then we should presumably return the number of commits on the branch
```

Comment:
I think it would be simpler to start with `int depth = 0;` and move `depth++;`  below conditional return block in while loop(line 72) and just return `new GitDescription(revObjectId.getName(), "root", depth);` here.

Reply:
I believe this is dealt with by my  latest commit - please re-review when you get time